### PR TITLE
[Core] Create Integration and Quadrature Points

### PR DESCRIPTION
**Pull request by tteschemacher**
Dear all,

this defines the interface to create integration points and with those quadrature point geometries.

Cheers!

### CODE REVIEW

File: kratos/integration/integration_point_utilities.cpp
Comment (tteschemacher) [inline kratos/integration/integration_point_utilities.cpp]: @pooyan-dadvand when I change the inner std::vector to array_1d I cannot initialize in that easy way. Any idea about that?

Comment (pooyan-dadvand) [reply kratos/integration/integration_point_utilities.cpp]: You are right! So you may use the `std::array<double,3>`

File: kratos/geometries/geometry.h
Comment (pooyan-dadvand) [inline kratos/geometries/geometry.h:1950]: Just for the record: Here a new IntegrationMethodInfo class will be passed as argument. But as the class definition is not ready yet. The addition will be postponed to a separate PR.

**Review by pooyan-dadvand — COMMENTED** [review]
It is ok for me. I would wait for @KratosMultiphysics/technical-committee to approve it.

**Comment by tteschemacher** [comment]
@KratosMultiphysics/technical-committee  do you agree with the additional features

**Review by RiccardoRossi — COMMENTED** [review]
@KratosMultiphysics/technical-committee is positive about the idea of the PR.

 @pooyan-dadvand will approve after reviewing

**Comment by tteschemacher** [comment]
Thank you @KratosMultiphysics/technical-committee and @RiccardoRossi for your review!

**Review by pooyan-dadvand — APPROVED** [review]
👍 

Please don't forget the IntegrationInfo class! 😉

**Comment by tteschemacher** [comment]
Thank you.

No worries about that I'm in desperate need of it. But I suppose it'll be a longer discussion. @

**Pull request merged**
